homeassistant/components/ssdp/__init__.py

```diff
@@ -85,6 +86,29 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+_SsdpServiceInfoBase = TypedDict(
```

Comment:
Can this be defined using normal class syntax?

Reply:
I tried (and struggled), but the key has dashes "x-homeassistant-matching-domains" and I couldn't find a normal class syntax. If you show me the syntax I'm happy to implement.

Reply:
Ah, of course. I think you can rename the key to use underscores; it's a field that Home Assistant is adding and has complete control over (name and contents), and the only user of it (`dlna_dmr`) uses the constant `ATTR_HA_MATCHING_DOMAINS` to access it.